api/src/main/java/org/apache/iceberg/types/Comparators.java

```diff
@@ -212,15 +223,18 @@ private NullsLast() {
 
     @Override
     public int compare(T o1, T o2) {
+      if (o1 == o2) {
+        return 0;
+      }
+
       if (o1 != null) {
         if (o2 != null) {
           return 0;
         }
         return -1;
-      } else if (o2 != null) {
-        return 1;
       }
-      return 0;
+
+      return 1;
```

Comment:
Like above, the `o1 == o2` case removes the possibility that `o2` is null. Otherwise, it would be equal to `o1`.